internal-services/catalog/pulp-push-disk-images-task.yaml

```diff
@@ -211,3 +239,120 @@ spec:
         pulp_push_wrapper --debug --source "${DISK_IMAGE_DIR}" --pulp-url "$PULP_URL" \
           --pulp-cert $PULP_CERT_FILE --pulp-key $PULP_KEY_FILE --udcache-url "$UDC_URL" \
           2> "$STDERR_FILE"
+
+        echo "Publishing to Developer Portal..."
+
+        python3 <<EOF
+        """
+        Run push-cgw-metadata command to push metadata to CGW:
+        1. Extract all components under contentGateway key from dataPath
+        2. Find all the files in contentDir that starts with the component name
+        4. Generate necessary metadata for each file
+        5. Dump the metadata to a YAML file
+        6. Run push-cgw-metadata to push the metadata
+        """
+        import os
+        import json
+        import yaml
+        import hashlib
+        import subprocess
+
+        WORKSPACE_DIR = os.path.dirname("/tmp")
+        METADATA_FILE_PATH = f"{WORKSPACE_DIR}/cgw_metadata.yaml"
+        RESULT_FILE_JSON_PATH=f"{WORKSPACE_DIR}/results.json"
+
+        os.makedirs(WORKSPACE_DIR, exist_ok=True)
+
+        productName = "$(params.productName)"
+        productCode = "$(params.productCode)"
+        productVersionName = "$(params.productVersionName)"
+        components = json.loads("$(params.components_json)")
+
+        content_list = os.listdir(DISK_IMAGE_DIR)
+
+        # Default values for each component,
+        # values from DATA_FILE takes presedence over these
+        default_values_per_component = {
+            'type': "FILE",
+            'shortURL': f"/cgw/{productCode}",
+            "hidden": False,
+            "invisible": False
+        }
+
+        def generate_download_url(file_name):
+            """
+            Generate a download URL in this format:
+            /content/origin/files/sha256/{checksum[:2]}{checksum}/{file_name}
+            """
+            prefix = "/content/origin/files/sha256"
+            sha256_hash = hashlib.sha256()
+            with open(CONTENT_DIR + "/" + file_name, "rb") as f:
```

Comment:
CONTENT_DIR is here too but I don't see it defined anywhere

Reply:
fixed